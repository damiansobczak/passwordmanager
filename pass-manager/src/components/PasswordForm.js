import React from "react";
import "./PasswordForm.scss";
import { getCurrentDate } from "../helpers";

class PasswordForm extends React.Component {
  passwordNameInput = React.createRef();
  passwordHashInput = React.createRef();
  form = React.createRef();
  submitPassword = e => {
    e.preventDefault();
    const currentDate = getCurrentDate();
    if (this.passwordNameInput.current.value && this.passwordNameInput.current.value) {
      const password = {
        name: this.passwordNameInput.current.value,
        password: this.passwordHashInput.current.value,
        date: currentDate
      };
      this.props.submitPassword(password);
      this.form.current.reset();
    }
  };

  render() {
    return (
      <div className="card">
        <div className="card-content">
          <div className="card-title">Dodaj hasło</div>
          <form onSubmit={this.submitPassword} ref={this.form}>
            <div className="row card-form-inputs">
              <div className="input-field col s5">
                <input type="text" className="validate" id="password-name" ref={this.passwordNameInput} />
                <label htmlFor="password-name">Strona</label>
              </div>
              <div className="input-field col s5">
                <input type="password" className="validate" id="password-hash" ref={this.passwordHashInput} />
                <label htmlFor="password-hash">Hasło</label>
              </div>
              <div className="input-field col s2">
                <button className="blue accent-4 accent-bluewaves-effect waves-light btn password-copy">Zapisz</button>
              </div>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

export default PasswordForm;
