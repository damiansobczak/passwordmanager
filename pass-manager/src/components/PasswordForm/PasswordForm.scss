.card-form-inputs {
  display: flex;
  align-items: center;
}
