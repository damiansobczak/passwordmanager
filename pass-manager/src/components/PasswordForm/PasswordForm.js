import React from "react";
import "./PasswordForm.scss";
import { getCurrentDate } from "../../helpers";
import PasswordChecker from "../PasswordChecker/PasswordChecker";

class PasswordForm extends React.Component {
  passwordNameInput = React.createRef();
  passwordHashInput = React.createRef();
  form = React.createRef();
  submitPassword = e => {
    e.preventDefault();
    const currentDate = getCurrentDate();
    if (this.passwordNameInput.current.value && this.passwordNameInput.current.value) {
      const password = {
        name: this.passwordNameInput.current.value,
        password: this.passwordHashInput.current.value,
        date: currentDate
      };
      this.props.submitPassword(password);
      this.form.current.reset();
    }
  };

  render() {
    return (
      <div className="card">
        <div className="card-content">
          <div className="card-title">Dodaj hasło</div>
          <form onSubmit={this.submitPassword} ref={this.form}>
            <div className="row card-form-inputs">
              <div className="input-field col s5">
                <input type="text" className="validate" id="password-name" ref={this.passwordNameInput} />
                <label htmlFor="password-name">Strona</label>
              </div>
              <div className="input-field col s5">
                <input type="password" className="validate" id="password-hash" ref={this.passwordHashInput} onChange={e => this.props.testPassword(e.target.value)} />
                <label htmlFor="password-hash">Hasło</label>
              </div>
              <div className="input-field col s2">
                <button className="blue accent-4 accent-bluewaves-effect waves-light btn password-copy">Zapisz</button>
              </div>
            </div>
            <PasswordChecker password={this.props.password} />
          </form>
        </div>
      </div>
    );
  }
}

export default PasswordForm;
