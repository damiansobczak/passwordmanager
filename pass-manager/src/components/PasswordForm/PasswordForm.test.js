import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import PasswordForm from "./PasswordForm";

describe("<PasswordForm />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<PasswordForm submitPassword={() => true} password="hashedpassword" testPassword={() => true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("mounts password checker component", () => {
    const wrapper = shallow(<PasswordForm submitPassword={() => true} />);
    expect(wrapper.find("PasswordChecker")).toHaveLength(1);
  });
});
