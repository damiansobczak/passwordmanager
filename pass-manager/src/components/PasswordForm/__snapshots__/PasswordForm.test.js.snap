// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordForm /> renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-content"
  >
    <div
      className="card-title"
    >
      Dodaj hasło
    </div>
    <form
      onSubmit={[Function]}
    >
      <div
        className="row card-form-inputs"
      >
        <div
          className="input-field col s5"
        >
          <input
            className="validate"
            id="password-name"
            type="text"
          />
          <label
            htmlFor="password-name"
          >
            Strona
          </label>
        </div>
        <div
          className="input-field col s5"
        >
          <input
            className="validate"
            id="password-hash"
            onChange={[Function]}
            type="password"
          />
          <label
            htmlFor="password-hash"
          >
            Hasło
          </label>
        </div>
        <div
          className="input-field col s2"
        >
          <button
            className="blue accent-4 accent-bluewaves-effect waves-light btn password-copy"
          >
            Zapisz
          </button>
        </div>
      </div>
      <div
        className="form card-form-inputs password-strength-meter active"
      >
        <div
          className="input-field col s12"
        >
          <div
            className="password-strength-meter-wrapper"
          >
            <progress
              className="password-strength-meter-progress strength-Good"
              max="4"
              value={3}
            />
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
